resolwe_bio/processes/reads_processing/bbduk.py

```diff
@@ -0,0 +1,1223 @@
+                default=-1,
+                description="If positive, discard reads with a larger X coordinate.",
+            )
+            y_max = IntegerField(
+                label="Maximum Y coordinate",
+                default=-1,
+                description="If positive, discard reads with a larger Y coordinate.",
+            )
+
+        class Complexity:
+            """Complexity parameters."""
+
+            entropy = FloatField(
+                label="Minimum entropy",
+                default=-1.0,
+                description="Set between 0 and 1 to filter reads with entropy below that value. "
+                "Higher is more stringent.",
+            )
+            entropy_window = IntegerField(
+                label="Length of sliding window used to calculate entropy",
+                default=50,
+                description="To use the sliding window set minimum entropy in range between 0.0 and 1.0.",
+            )
+            entropy_k = IntegerField(
+                label="Length of kmers used to calcuate entropy",
+                default=5,
+            )
+            entropy_mask = BooleanField(
+                label="Mask low-entropy parts of sequences with N instead of discarding",
+                default=False,
+            )
+            min_base_frequency = IntegerField(
+                label="Minimum base frequency",
+                default=0,
+            )
+
+        class Fastqc:
+            """FastQC parameters."""
+
+            nogroup = BooleanField(
+                label="Disable grouping of bases for reads >50bp",
+                default=False,
+                description="All reports will show data for every base in the read. Using this option "
+                "will cause fastqc to crash and burn if you use it on really long reads.",
+            )
+
+        reference = GroupField(Reference, label="Reference", hidden="!show_advanced")
+
+        processing = GroupField(
+            Processing, label="Processing parameters", hidden="!show_advanced"
+        )
+
+        operations = GroupField(
+            Operations,
+            label="Trimming, filtering and masking parameters.",
+            hidden="!show_advanced",
+        )
+
+        header_parsing = GroupField(
+            HeaderParsing, label="Header-parsing parameters", hidden="!show_advanced"
+        )
+
+        complexity = GroupField(
+            Complexity, label="Complexity parameters", hidden="!show_advanced"
+        )
+
+        fastqc = GroupField(Fastqc, label="FastQC parameters", hidden="!show_advanced")
+
+    class Output:
+        """Output fields."""
+
+        fastq = ListField(FileField(), label="Remaining reads")
+        statistics = ListField(FileField(), label="Statistics")
+        fastqc_url = ListField(FileHtmlField(), label="Quality control with FastQC")
+        fastqc_archive = ListField(FileField(), label="Download FastQC archive")
+
+    def run(self, inputs, outputs):
```

Comment:
Consider adding `self.progress()` throughout the process to show progress where this functionality may be used.